examples/nary.mcl

```diff
@@ -0,0 +1,49 @@
+(* main body *)
+(
+  fun (Nary : forall (n : Nat) -> Type@0)
```

Comment:
what's the return type for this function? no need to specify it?

Reply:
oh! this whole this is a single definition!

Reply:
Yeah, we don't have top-level definitions yet.